text/src/autogluon/text/automm/predictor.py

```diff
@@ -578,91 +756,127 @@ def _fit(
             )
 
         if trainer.global_rank == 0:
-            top_k_avg_ckpt_path = os.path.join(save_path, "model.ckpt")
-            # Obtain the best K models sorted from the best performance to the worst performance
-            best_k_models_sorted = sorted(list(checkpoint_callback.best_k_models.items()),
-                                          key=lambda ele: ele[1], reverse=(minmax_mode == MAX))
-            best_k_models_path = [ele[0] for ele in best_k_models_sorted]
-
-            # TODO(?) Currently, all top-K checkpoints are loaded to the main memory.
-            #  If the checkpoints are really large (e.g., super-giant backbones), this operation can take a lot of
-            #  CPU memory and may potentially trigger memory issue. Consider to optimize that using the
-            #  offload strategy.
-            all_state_dicts = gather_top_k_ckpts(
-                ckpt_dir=save_path,
-                ckpt_paths=best_k_models_path,
+            self._top_k_average(
+                model=model,
+                save_path=save_path,
+                checkpoint_callback=checkpoint_callback,
+                minmax_mode=minmax_mode,
+                is_distill=is_distill,
+                config=config,
+                val_df=val_df,
+                validation_metric_name=validation_metric_name,
+            )
+        else:
+            sys.exit(
+                f"Training finished, exit the process with global_rank={trainer.global_rank}..."
             )
```

Comment:
Is this happening in a sub-process? Is there a clean way to exit the process without calling `sys.exit`?

Reply:
For `ddp` and `ddp_spawn`, each GPU has its process. Refer to here: https://pytorch-lightning.readthedocs.io/en/1.4.3/advanced/multi_gpu.html#distributed-data-parallel